ngraph/core/src/op/topk.cpp

```diff
@@ -276,6 +277,11 @@ void op::v1::TopK::validate_and_infer_types()
     NODE_VALIDATION_CHECK(
         this, k_partial_shape.rank().compatible(0), "The 'K' input must be a scalar.");
 
+    NODE_VALIDATION_CHECK(this,
```

Comment:
Shouldn't `index_element_type` be compared to `i32` and `i64` instead of "i32" and ""i64? As in constructor it is of `element::Type` type

Reply:
Yes, good point.

Reply:
You are changing logic for an existing opset1 operation and it's prohibited because of potential backward compatibility issues. Also I see that `v1::TopK::validate_and_infer_types` is used in v3 TopK version. But it is also not new operation so we can't put more restrictions to it's attribute for the same reason. @lazarevevgeny please correct me if I'm wrong.

Also it's interesting that opset1 version of TopK according to the spec https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/sort/TopK_1.md doesn't have index_element_type attribute. @lazarevevgeny should we change the spec in this case?

https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/sort/TopK_3.md 


So at least you need to check that

Reply:
`index_element_type` should not be added to the `visit_attributes` of the `v1::TopK`. This attribute was added to `v3::TopK`. Thanks, @GlebKazantaev!

The fact that we add more restrictions to the existing ops is acceptable because these restrictions correspond to our specification. Also, we already added many restrictions to other existing ops as part of the operations revise :)

Reply:
Thanks for clarification. I should remove index_element_type from visit_attributes, but the validation check of index_element_type should stay, right?

Reply:
The attribute `index_element_type` should not be a part of `v1::TopK` at all. So it should not be checked in `v1::TopK` but should be checked in `v3::TopK`.

Reply:
It turned out that according to https://github.com/NervanaSystems/ngraph/blame/master/src/ngraph/op/topk.hpp index_element_type should also be an attribute for v1 :: TopK, so I updated the spec instead of removing index_element_type from visit_attribute and its node_validation_check